src/qtcore/qml/QMLBaseObject.js

```diff
@@ -14,6 +14,10 @@ function QMLBaseObject(meta) {
     engine.completedSignals.push(this.Component.completed);
     this.completed = this.Component.completed;
 
+    this.Component.destruction = Signal([]);
+    //engine.destructionSignals.push(this.Component.destruction); would we need this?
```

Comment:
@ChALkeR now I thinked again and it seems to be obvious, when we will decide to do graceful shutdown. So you might just remove that line for now.